src/Nethermind/Nethermind.Core/Crypto/KeccakHash.cs

```diff
@@ -104,130 +100,98 @@ private static void KeccakF(Span<ulong> st)
             ulong ema, eme, emi, emo, emu;
             ulong esa, ese, esi, eso, esu;
 
-            {
-                // Access last element to perform range check once
-                // and not for every ascending access
-                _ = st[24];
-            }
-            aba = st[0];
-            abe = st[1];
-            abi = st[2];
-            abo = st[3];
-            abu = st[4];
-            aga = st[5];
-            age = st[6];
-            agi = st[7];
-            ago = st[8];
-            agu = st[9];
-            aka = st[10];
-            ake = st[11];
-            aki = st[12];
-            ako = st[13];
-            aku = st[14];
-            ama = st[15];
-            ame = st[16];
-            ami = st[17];
-            amo = st[18];
-            amu = st[19];
-            asa = st[20];
-            ase = st[21];
-            asi = st[22];
-            aso = st[23];
-            asu = st[24];
-
             for (int round = 0; round < ROUNDS; round += 2)
             {
                 //    prepareTheta
-                bCa = aba ^ aga ^ aka ^ ama ^ asa;
-                bCe = abe ^ age ^ ake ^ ame ^ ase;
-                bCi = abi ^ agi ^ aki ^ ami ^ asi;
-                bCo = abo ^ ago ^ ako ^ amo ^ aso;
-                bCu = abu ^ agu ^ aku ^ amu ^ asu;
-
-                //thetaRhoPiChiIotaPrepareTheta(round  , A, E)
-                da = bCu ^ RotateLeft(bCe, 1);
+                bCa = st.aba();
+                bCe = st.abe();
+                bCi = st.abi();
+                bCo = st.abo();
+                bCu = st.abu();
+
+                bCa ^= st.aga();
+                bCe ^= st.age();
+                bCi ^= st.agi();
+                bCo ^= st.ago();
+                bCu ^= st.agu();
+
+                bCa ^= st.aka();
+                bCe ^= st.ake();
+                bCi ^= st.aki();
+                bCo ^= st.ako();
+                bCu ^= st.aku();
+
+                bCa ^= st.ama();
+                bCe ^= st.ame();
+                bCi ^= st.ami();
+                bCo ^= st.amo();
+                bCu ^= st.amu();
+
+                bCa ^= st.asa();
+                bCe ^= st.ase();
+                bCi ^= st.asi();
                 de = bCa ^ RotateLeft(bCi, 1);
+                bCo ^= st.aso();
                 di = bCe ^ RotateLeft(bCo, 1);
-                @do = bCi ^ RotateLeft(bCu, 1);
+                bCu ^= st.asu();
```

Comment:
in many steps we have bCx and dX steps mixed, maybe it would be more readable if we unmix them. I think they are not dependent on each other.